pyomo/contrib/doe/tests/test_doe_FIM_metrics.py

```diff
@@ -0,0 +1,73 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2025
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+from pyomo.common.dependencies import (
+    numpy as np,
+    numpy_available,
+)
+
+import pyomo.common.unittest as unittest
+from pyomo.contrib.doe.doe import (
+    _SMALL_TOLERANCE_IMG,
+    _compute_FIM_metrics,
+)
+
+
+@unittest.skipIf(not numpy_available, "Numpy is not available")
+class TestComputeFIMMetrics(unittest.TestCase):
+    def test_compute_FIM_metrics(self):
+        # Create a sample Fisher Information Matrix (FIM)
+        FIM = np.array([[10, 2], [2, 3]])
+
+        det_FIM, trace_FIM, E_vals, E_vecs, D_opt, A_opt, E_opt, ME_opt = (
+            _compute_FIM_metrics(FIM)
+        )
+
+        # expected results
+        det_expected = np.linalg.det(FIM)
+        D_opt_expected = np.log10(det_expected)
+
+        trace_expected = np.trace(FIM)
+        A_opt_expected = np.log10(trace_expected)
+
+        E_vals_expected, E_vecs_expected = np.linalg.eig(FIM)
+        min_eigval = np.min(E_vals_expected.real)
+        if min_eigval <= 0:
+            E_opt_expected = np.nan
+        else:
+            E_opt_expected = np.log10(min_eigval)
+
+        cond_expected = np.linalg.cond(FIM)
+
+        ME_opt_expected = np.log10(cond_expected)
```

Comment:
You typically want to hard-code the expected results in the test instead of copying the function code here. This isn't really testing anything except that the code is the same in both places. If the numpy behavior ever changed resulting in different values, this test would not catch it and would always pass.